spark/src/main/scala/geotrellis/spark/io/hadoop/HadoopRDDWriter.scala

```diff
@@ -17,21 +17,47 @@
 package geotrellis.spark.io.hadoop
 
 import geotrellis.spark._
-import geotrellis.spark.util._
-import geotrellis.spark.partition._
-import geotrellis.spark.io.index._
+import geotrellis.spark.io._
 import geotrellis.spark.io.avro._
 import geotrellis.spark.io.avro.codecs._
+import geotrellis.spark.io.hadoop.formats.FilterMapFileInputFormat
+import geotrellis.spark.io.index._
+import geotrellis.spark.partition._
+import geotrellis.spark.util._
+import geotrellis.spark.util.KryoWrapper
 import geotrellis.util.LazyLogging
 
+import org.apache.avro.Schema
+import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.hadoop.io._
-import org.apache.hadoop.conf.Configuration
 import org.apache.spark.rdd._
 
 import scala.reflect._
+import scala.collection.mutable
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+import org.apache.hadoop.conf.Configuration
+
 
 object HadoopRDDWriter extends LazyLogging {
+  /** Index innterval at which map files should store an offset into sequence file.
+    * This value is picked as a compromize between in-memory footprint and IO cost of retreiving a single record.
+    */
+  final val DefaultIndexInterval = 4
+
+  // From https://github.com/apache/spark/blob/3b049abf102908ca72674139367e3b8d9ffcc283/core/src/main/scala/org/apache/spark/util/SerializableConfiguration.scala
+  private class SerializableConfiguration(@transient var value: Configuration) extends Serializable {
```

Comment:
Can we put it into `util` package? It could be reused by RF and in my COGs layer support.

* [RF](https://github.com/raster-foundry/raster-foundry/blob/develop/app-backend/batch/src/main/scala/com/azavea/rf/batch/util/HadoopConfiguration.scala) // checked by time!
* [COGs](https://github.com/pomadchin/geotrellis/blob/feature/cog-layer/spark/src/main/scala/geotrellis/spark/util/HadoopConfiguration.scala)

I suggest to use [this](https://github.com/pomadchin/geotrellis/blob/feature/cog-layer/spark/src/main/scala/geotrellis/spark/util/HadoopConfiguration.scala) though not sure about necessity in `@transient`

Reply:
I wonder if a lack of `Serializable` trait has ever caused so much grief as in `Configuration`

Reply:
Can't put it into `geotrellis.util` on account of the missing hadoop dependency but it has a home at `geotrellis.spark.util`

Reply:
yes, that's what I meant; thanks!

Reply:
> I wonder if a lack of `Serializable` trait has ever caused so much grief as in `Configuration`

`CRS` not being `Product` has a similar feel.